### ISSUE TITLE
[Programmatic Access – Make Code - Create GitHub Repository] Ensures every form element has a label

### ISSUE THREAD

**Issue by Shalini3457** [microsoft/pxt-microbit#3758]
User Experience:
If instructions or labels does not clearly identify the controls in a form, the users will not know what input data is expected. The pages will look cluttered with unnecessary information and people with disabilities will miss important cues and instructions

Note: User credentials should NOT be included in the bug.

Repro Steps:

1. Launch application URL
2. Home page will open.
3. Navigate to "Import" button and select it.
4. Navigate through the controls appearing under "Import" dialog.
5. Navigate to "Your GitHub Repo" option and Select it.
6. Navigate and select "Sign In" control under Sign in with GitHub and navigate to all the controls of dialogue "Clone or create your own GitHub repo
7. Navigate and select "Create New" control under "Clone or create your own GitHub repo" and navigate to all the controls of dialogue "Create GitHub Repository
8. Verify the issue.

Actual Result:
Label is not properly defined under 'Create GitHub Repository' pop-up.

Expected Result:
Label should be properly define for  under 'Create GitHub Repository' pop-up.

Note: 
Same issue exists under 'New branch name' pop up.

Issue: Ensures every form element has a label (label - https://dequeuniversity.com/rules/axe/4.0/label?application=msftAI)

Target application: Microsoft MakeCode for micro:bit - https://makecode.microbit.org/#

Element path:  .fluid[type="text"][value=""]

Snippet: <input class="fluid" type="text" placeholder="MakeCode extension for my gadget" autocomplete="off" autocorrect="off" autocapitalize="off" value="">

How to fix:
Fix any of the following:
aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Form element does not have an implicit (wrapped) <label>
  Form element does not have an explicit <label>
  Element has no title attribute or the title attribute is empty
  Element's default semantics were not overridden with role="none"
  Element's default semantics were not overridden with role="presentation"

This accessibility issue was found using Accessibility Insights for Web 2.22.0 (axe-core 4.0.2), a tool that helps find and fix accessibility issues. Get more information & download this tool.

Note:
Same issue exists on Firefox & Chrome.

WCAG Reference:
[WCAG 1.3.1 - Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)

Test Environments:
Version: 20H2 (OS Build: 19042.508)
Browser: Edge chromium DEV Build (Version 87.0.654.0), Firefox(75.0), Chrome
Tool: Accessibility insights for web
Screen Reader: Narrator, NVDA, JAWS

**Comment by Shalini3457** [comment]
Labels:
#Accessibility; #A11ySev2; #HCL; #HCL-MakeCode-New; #WCAG1.3.1; #Win10-Edge(Chromium);

![Uploading WcAg1.3.1_ Ensures every form element has a label.png…]()
